src/common.ts

```diff
@@ -24,4 +25,17 @@ export class Common {
       });
     }
   }
+  public static async getLoginUrl(instanceUrl: Optional<string>): Promise<Optional<string>> {
+    if (instanceUrl) {
+      return instanceUrl;
+    }
+    try {
+      const project = await SfdxProject.resolve();
+      const projectJson = await project.resolveProjectConfig();
+      return getString(projectJson, 'sfdcLoginUrl', SfdcUrl.PRODUCTION);
+    } catch (err) {
```

Comment:
would it be useful to add some logging around here? Or is the instanceUrl already being logged from something else?

Reply:
I will add some logging to record how the login URL resolves.